estate/models/estate_offer.py

```diff
@@ -0,0 +1,63 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+
+
+class EstateOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "property offer"
+
+    price = fields.Float(string='Price')
+    partner_id = fields.Many2one('res.partner', required=True, string='Partner')
+    property_id = fields.Many2one('estate.property', required=True, string='Property')
+    date_creation = fields.Date(readonly=True, default=fields.Date.today)
+    status = fields.Selection(
+        string='Status',
+        copy=False,
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused')],
+    )
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        string='Deadline', compute="_compute_deadline", inverse="_inverse_deadline"
+    )
+
+    _check_price = models.Constraint(
+        'CHECK(price >= 0)', 'The offer price must be positive'
+    )
+
+    @api.depends('validity')
+    def _compute_deadline(self):
+        for record in self:
```

Comment:
```suggestion
        for estate in self:
```
We more and more try to avoid generic "record" naming in those cases to make it more readable.
This way even when reading a snippet we know what kind of records we're looping on.